src/pip/_internal/cache.py

```diff
@@ -148,3 +153,50 @@ def get(self, link, package_name):
             return link
 
         return self._link_for_candidate(link, min(candidates)[1])
+
+
+class EphemWheelCache(SimpleWheelCache):
+    """A SimpleWheelCache that creates it's own temporary cache directory
+    """
+
+    def __init__(self, format_control):
+        self._temp_dir = TempDirectory(kind="ephem-wheel-cache")
+        self._temp_dir.create()
+
+        super(EphemWheelCache, self).__init__(
+            self._temp_dir.path, format_control
+        )
+
+    def cleanup(self):
+        self._temp_dir.cleanup()
+
+
+class WheelCache(Cache):
+    """Wraps EphemWheelCache and SimpleWheelCache into a single Cache
+
+    This Cache allows for gracefully degradation, using the ephem wheel cache
+    when a certain link is not found in the simple wheel cache first.
+    """
+
+    def __init__(self, cache_dir, format_control):
+        super(WheelCache, self).__init__(
+            cache_dir, format_control, {'binary'}
+        )
+        self._wheel_cache = SimpleWheelCache(cache_dir, format_control)
+        self._ephem_cache = EphemWheelCache(format_control)
```

Comment:
Would it be worth building the ephemeral wheel cache lazily? The majority of installs won't be installing from git or a local directory, so why waste time creating and deleting a temporary directory that we never use?

This may be a premature optimisation, of course. Feel free to take the view that we can do this later if it proves to be a problem.

Reply:
```sh
$ python -c "
import time
import pip._internal.utils.temp_dir as tmp
t = tmp.TempDirectory()
print(time.time())
t.create()
t.cleanup()
print(time.time())"
1507128804.431749
1507128804.43445
```

At about 0.003 seconds, I don't think it's a major issue. Even if it's 10x as costly, I'm tempted to say it's not worth it to optimise.

Reply:
OK, agreed. Thanks for testing this. I was concerned that Windows might be slower (not an uncommon problem...) but I ran your test script on my machine, and it's fine. I'm convinced - leave it as it is.